### ISSUE TITLE
Handle escape character without escaped character in searches

### ISSUE THREAD

**Issue by ocrasborn** [Signbank/Global-signbank#714]
If a user uses \ in one of the regular expression search fields at the end of a string (as a typo, typically, because it's close to the enter key on many keyboard layouts), Django presents a nasty looking error. It would be nicer to simply ignore the backslash (and thus not bother the user with a friendly error message either).

**Comment by susanodd** [comment]
I can work on this.

It seems it's only on the menu bar search fields, is that correct?

**Comment by susanodd** [comment]
Okay, I've got it fixed.

I first tried to use some mappings to make such a string safe. But that didn't work because it's not a control character or an escape character. If you type in "\ " it doesn't crash.

I revised the code as you suggest to remove (ignore) an ending "\" in the search term.

**Comment by susanodd** [comment]
This may not work any more for regex.

**Comment by susanodd** [comment]
The code that resolves this issue is the mysterious `strip_control_characters` function.

(This was removed due to confusion about why it was in the code.)

**Comment by susanodd** [comment]
Yes, it does this now because it's a poorly formed regex and is thus missing backslashes.
So we niks this issue?
